discretize/CurvilinearMesh.py

```diff
@@ -41,28 +115,48 @@ class CurvilinearMesh(BaseRectangularMesh, DiffOperators, InnerProducts, CurviVi
 
     _meshType = 'Curv'
 
-    def __init__(self, nodes):
-        assert type(nodes) == list, ("'nodes' variable must be a list of "
-                                     "np.ndarray")
-        assert len(nodes) > 1, "len(node) must be greater than 1"
+    nodes = properties.List(
+        "List of arrays describing the node locations",
+        Array(
+            "node locations in a dimension",
+            shape='*'
+        ),
```

Comment:
min=2 and max=3 length on the list?

Reply:
This is an array, so we can't, but I will implement the Union Property as suggested by @fwkoch